packages/agoric-cli/lib/start.js

```diff
@@ -111,6 +204,283 @@ export default async function startMain(progname, rawArgs, powers, opts) {
+            `--rm`,
+            ...dockerArgs,
+            `-it`,
+            IMAGE,
+            `--home=/usr/src/dapp/${agServer}`,
+            ...args,
+          ],
+          spawnOpts,
+        );
+    }
+
+    if (!(await exists(agServer))) {
+      const status = await chainSpawn([
+        'init',
+        'local-chain',
+        `--chain-id=${CHAIN_ID}`,
+      ]);
+      if (status) {
+        return status;
+      }
+    }
+
+    // Get or create the essential addresses.
+    const addrs = {};
+    for (const keyName of ['delegate0', 'provision']) {
+      /* eslint-disable no-await-in-loop */
+      let capret = showKey(keyName);
+      if (await capret[0]) {
+        const status = await keysSpawn([
+          'keys',
+          'add',
+          keyName,
+          '--keyring-backend=test',
+        ]);
+        if (status) {
+          return status;
+        }
+        capret = showKey(keyName);
+        const status2 = await capret[0];
+        if (status2) {
+          return status2;
+        }
+      }
+      addrs[keyName] = capret[1].trimRight();
+      /* eslint-enable no-await-in-loop */
+    }
+
+    const genfile = `${agServer}/config/genesis.json`;
+    if (!(await exists(`${genfile}.stamp`))) {
+      let status;
+      await chainSpawn([
+        'add-genesis-account',
+        addrs.provision,
+        PROVISION_PASSES,
+      ]);
+      if (status) {
+        return status;
+      }
+      await chainSpawn([
+        'add-genesis-account',
+        addrs.delegate0,
+        DELEGATE0_STAKE,
+      ]);
+      if (status) {
+        return status;
+      }
+      const keysHome = opts.sdk
+        ? `_agstate/keys`
+        : `/usr/src/dapp/_agstate/keys`;
+      status = await chainSpawn([
+        'gentx',
+        `--home-client=${keysHome}`,
+        '--keyring-backend=test',
+        '--name=delegate0',
+        `--amount=${DELEGATE0_STAKE}`,
+      ]);
+      if (status) {
+        return status;
+      }
+      status = await chainSpawn(['collect-gentxs']);
+      if (status) {
+        return status;
+      }
+      status = await chainSpawn(['validate-genesis']);
+      if (status) {
+        return status;
+      }
+      status = await fs.writeFile(`${genfile}.stamp`, Date.now());
+      if (status) {
+        return status;
+      }
+    }
+
+    // Complete the genesis file and launch the chain.
+    await finishGenesis(genfile);
+    await finishConfig(`${agServer}/config/config.toml`, portNum);
+    return chainSpawn(
+      ['start', '--pruning=nothing'],
+      {
+        env: { ...pspawnEnv, ROLE: 'two_chain' },
+      },
+      // Accessible via either localhost or host.docker.internal
+      [`--publish=${portNum}:${portNum}`, `--name=agoric/n0`],
+    );
+  }
+
+  async function startLocalSolo(profileName, startArgs, popts) {
+    const IMAGE = `agoric/agoric-sdk`;
+
+    const portNum = startArgs[0] === undefined ? PORT : startArgs[0];
+    if (`${portNum}` !== `${Number(portNum)}`) {
```

Comment:
you might want to `Nat` this too, since hyphens are also being used as argument indicators here.. I can imagine someone calling it with `-1234` instead of `1234` and causing confusion because it's still a legitimate Number

Reply:
Actually, the `commander` library insists on parsing all the options up-front.  But I could agree that `Nat` would be useful.